types/superagent/index.d.ts

```diff
@@ -102,7 +103,7 @@ declare namespace request {
       send(): this;
       set(field: string, val: string): this;
       set(field: Object): this;
-      timeout(ms: number): this;
+      timeout(ms: number | Object | { response: number, read: number, deadline: number }): this;
```

Comment:
@mhegazy I copied the Object param from superagent signature, but it's really not necessary. Updated the branch.